back/app/services/side_fx_group_service.rb

```diff
@@ -13,7 +13,13 @@ def after_create(group, user)
   def before_update(group, user); end
 
   def after_update(group, user)
-    LogActivityJob.perform_later(group, 'changed', user, group.updated_at.to_i)
+    LogActivityJob.perform_later(
+      encode_frozen_resource(group),
+      'changed',
+      user,
+      Time.now.to_i,
+      payload: { changes: group.previous_changes }
+    )
```

Comment:
I noticed that in [SideFxIdeaService.after_update](https://github.com/CitizenLabDotCo/citizenlab/blob/bd2fb192eb63e7611723aac5443064bd80e52095/back/app/services/side_fx_idea_service.rb#L32) we use a number of different `LogActivityJob`s, each logging changes to specific `Idea` attribute. What are the advantages of this approach over the logging of the more general `group.previous_changes` I am thinking about using here?

Reply:
Main reason why we have some specific `action` values for a change has to do with the handling of these activities downstream.

For example, we have RabbitMQ listening to these activities. A consumer might only be interested in certain changes. In RabbitMQ, we can choose to subscribe to specific events only, and then we'd only receive data and execute code when these specific events happen. Without those, we would have to listen to every individual event, read the payload and then decide whether we continue or not.

For analytics purposes, it can also be nice to have separate events for meaningful things that happen in the application. Some tools might not make it possible or easy to filter things by nested payload variables.

And finally, depending on the changed data attributes means that we are coupling our database structure to downstream logic of activities. Suppose we want to rename a certain database field, then we'd have to change all listeners. If we make explicit events with explicit payloads, we wouldn't have to.

All that being said, both approaches don't per se exclude each other: We can still have explicit activities for major, meaningful domain events, while at the same time assigning the lower-level database changes to the payload. It can be convenient and practical to have these, so I'm in favor of adding the changes more consistently.